Index: aptdaemon/aptdaemon/worker/aptworker.py
===================================================================
--- aptdaemon.orig/aptdaemon/worker/aptworker.py
+++ aptdaemon/aptdaemon/worker/aptworker.py
@@ -113,8 +113,10 @@ def trans_only_installs_pkgs_from_high_t
     # paranoia(2): we must want to install something
     if not trans.packages[PKGS_INSTALL]:
         return False
+    # we only care about the name, not the version
+    pkgs = [pkg.split("=")[0] for pkg in trans.packages[PKGS_INSTALL]]
     # if the install packages matches the whitelisted set we are good
-    return set(trans.packages[PKGS_INSTALL]) == set(trans.high_trust_packages)
+    return set(pkgs) == set(trans.high_trust_packages)
 
 
 def read_high_trust_repository_dir(whitelist_cfg_d):
