Description: Set the minimum scrolled window height
Author: Martin Wimpress
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/update-manager/+bug/1623856

=== modified file 'aptdaemon/gtk3widgets.py'
--- a/aptdaemon/gtk3widgets.py
+++ b/aptdaemon/gtk3widgets.py
@@ -338,6 +338,7 @@
         self.download_scrolled.set_policy(Gtk.PolicyType.NEVER,
                                           Gtk.PolicyType.AUTOMATIC)
         self.download_scrolled.add(self.download_view)
+        self.download_scrolled.set_min_content_height(200)
         hbox = Gtk.HBox()
         hbox.pack_start(self.download_scrolled, True, True, 0)
         if self.terminal:
