Index: aptdaemon-1.1.1+bzr982/aptdaemon/gtk3widgets.py
===================================================================
--- aptdaemon-1.1.1+bzr982.orig/aptdaemon/gtk3widgets.py
+++ aptdaemon-1.1.1+bzr982/aptdaemon/gtk3widgets.py
@@ -1066,7 +1066,7 @@ class AptConfigFileConflictDialog(_Expan
                   "the configuration file?") % from_path
         msg = _("If you don't know why the file is there already, it is "
                 "usually safe to replace it.")
-        scrolled = Gtk.ScrolledWindow()
+        scrolled = Gtk.ScrolledWindow(hexpand=True, vexpand=True)
         scrolled.set_policy(Gtk.PolicyType.AUTOMATIC, Gtk.PolicyType.AUTOMATIC)
         scrolled.set_shadow_type(Gtk.ShadowType.ETCHED_IN)
         self.diffview = DiffView()
