Index: aptdaemon-1.1.1+bzr982/aptdaemon/gtk3widgets.py
===================================================================
--- aptdaemon-1.1.1+bzr982.orig/aptdaemon/gtk3widgets.py
+++ aptdaemon-1.1.1+bzr982/aptdaemon/gtk3widgets.py
@@ -395,6 +395,7 @@ class AptTerminal(Vte.Terminal):
         self._master, self._slave = pty.openpty()
         self._ttyname = os.ttyname(self._slave)
         self.set_size(80, 24)
+        self.set_vexpand(True)
         self.set_pty(Vte.Pty.new_foreign_sync(self._master))
         if transaction is not None:
             self.set_transaction(transaction)
