From: Michael Vogt <mvo@ubuntu.com>
Subject: Remove API that is no longer available in PK 1.1.1

---
 aptdaemon/worker/pkworker.py |    6 ------
 1 file changed, 6 deletions(-)

--- a/aptdaemon/worker/pkworker.py
+++ b/aptdaemon/worker/pkworker.py
@@ -172,8 +172,6 @@
                                         pk.RoleEnum.GET_UPDATE_DETAIL,
                                         pk.RoleEnum.GET_PACKAGES,
                                         pk.RoleEnum.GET_DETAILS,
-                                        pk.RoleEnum.GET_DEPENDS,
-                                        pk.RoleEnum.GET_REQUIRES,
                                         pk.RoleEnum.SEARCH_NAME,
                                         pk.RoleEnum.SEARCH_DETAILS,
                                         pk.RoleEnum.SEARCH_GROUP,
@@ -289,10 +287,6 @@
             self.search_details(trans, **trans.kwargs)
         elif trans.pktrans.role == pk.RoleEnum.SEARCH_FILE:
             self.search_files(trans, **trans.kwargs)
-        elif trans.pktrans.role == pk.RoleEnum.GET_DEPENDS:
-            self.get_depends(trans, **trans.kwargs)
-        elif trans.pktrans.role == pk.RoleEnum.GET_REQUIRES:
-            self.get_requires(trans, **trans.kwargs)
         elif trans.pktrans.role == pk.RoleEnum.GET_DETAILS:
             self.get_details(trans, **trans.kwargs)
         elif trans.pktrans.role == pk.RoleEnum.DOWNLOAD_PACKAGES:
