=== modified file 'aptdaemon/progress.py'
--- a/aptdaemon/progress.py	2014-02-07 20:40:15 +0000
+++ b/aptdaemon/progress.py	2014-04-11 11:58:33 +0000
@@ -395,6 +395,9 @@
         self.last_activity = 0
         self.child_pid = 0
         self.status_parent_fd, self.status_child_fd = os.pipe()
+        if hasattr(os, "set_inheritable"):
+            os.set_inheritable(self.status_parent_fd, True)
+            os.set_inheritable(self.status_child_fd, True)
         self.output = ""
         self._line_buffer = ""
 

